### DISCUSSION TITLE
presenting/

### DISCUSSION THREAD

**Discussion by giscus** [zmike/zmike.github.io#15] [Announcements]
# presenting/

When perf Is Too Slow

https://www.supergoodcode.com/presenting/

**Comment by jams3223** [comment]
Keep on going, i believe in you !

**Comment by ermarch** [comment]
Just a thought, and not sure if it works in case of Zink:
Is it an option to call vkGetPhysicalDeviceSurfaceCapabilitiesKHR at the end of the sequence (when there is nothing else to do) to have it's information available for the next round? Trying to take advantage of the moment where the GPU is busy rendering and the CPU has some spare cycles.

**Reply by zmike** [reply]
It's irrelevant when it gets called since the call is (currently) blocking.

**Reply by ermarch** [reply]
That doesn't help. I mentioned it because it worked for me for a particular application.

**Comment by orowith2os** [comment]
What's the performance like on *Wayland*, is what I'm wondering. The API is simpler, smaller, and way more efficient, wouldn't that give you a considerable performance boost?

**Reply by zmike** [reply]
glmark2 is an X11 application.

**Reply by qwertychouskie** [reply]
Apparently `glmark2-wayland` is a thing, I'd be curious to see the performance difference there.

**Reply by orowith2os** [reply]
Testing Xwayland could also be worth a try, alongside `glmark2-wayland`.
